pgmpy/factors/Factor.py

```diff
@@ -314,7 +315,26 @@ def product(self, *factors, n_jobs=1):
         OrderedDict([('x1', ['x1_0', 'x1_1']), ('x2', ['x2_0', 'x2_1', 'x2_2']),
                 ('x3', ['x3_0', 'x3_1']), ('x4', ['x4_0', 'x4_1'])])
         """
-        return factor_product(self, *factors, n_jobs=n_jobs)
+        if isinstance(factors[0], Number):
+            return Factor(self.variables, self.cardinality, np.array([i * factors[0] for i in self.values]))
+        else:
+            return factor_product(self, *factors, n_jobs=n_jobs)
+
+    def sum(self, factor, n_jobs=1):
+        """
+        Returns a new factors instance after addition with factor
+
+        Parameters
+        ----------
+        factor : factors
+            The factor to be added
+        """
+        """
+        :param factor:
```

Comment:
@ankurankan : okay.
I needed an add functionality for factors. Any other work around or should i just wait for the reworked Factor?